contracts/GroupHealth.sol

```diff
@@ -0,0 +1,297 @@
+     */
+    function isValidGroup(address group) public view returns (bool) {
+        IValidators validators = getValidators();
+
+        // add check if group is !registered
+        if (!validators.isValidatorGroup(group)) {
+            return false;
+        }
+
+        (address[] memory members, , , , , uint256 slashMultiplier, ) = validators
+            .getValidatorGroup(group);
+
+        // check if group has no members
+        if (members.length == 0) {
+            return false;
+        }
+        // check for recent slash
+        if (slashMultiplier < 10**24) {
+            return false;
+        }
+        // check that at least one member is elected.
+        for (uint256 i = 0; i < members.length; i++) {
+            if (isGroupMemberElected(members[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Returns expected Celo amount voted for by Account contract
+     * vs actual amount voted for by Acccount contract
+     * @param group The group.
+     */
+    function getExpectedAndRealCeloForGroup(address group) public view returns (uint256, uint256) {
+        bool isSpecificGroupStrategy = specificGroupStrategy.isSpecificGroupStrategy(group);
+        bool isActiveGroup = defaultStrategy.groupsContain(group);
+        uint256 realCelo = account.getCeloForGroup(group);
+
+        if (!isSpecificGroupStrategy && !isActiveGroup) {
+            return (0, realCelo);
+        }
+
+        if (isSpecificGroupStrategy && !isActiveGroup) {
+            return (
+                manager.toCelo(specificGroupStrategy.getTotalStCeloVotesForStrategy(group)),
+                realCelo
+            );
+        }
+
+        if (!isSpecificGroupStrategy && isActiveGroup) {
+            uint256 stCeloSupply = stakedCelo.totalSupply();
+            uint256 stCeloInDefaultStrategy = stCeloSupply -
+                specificGroupStrategy.getTotalStCeloInSpecificGroupStrategies();
+            uint256 supposedStCeloInActiveGroup = stCeloInDefaultStrategy /
+                defaultStrategy.getGroupsLength();
+
+            return (manager.toCelo(supposedStCeloInActiveGroup), realCelo);
+        }
+
+        if (isSpecificGroupStrategy && isActiveGroup) {
+            uint256 stCeloSupply = stakedCelo.totalSupply();
+            uint256 stCeloInDefaultStrategy = stCeloSupply -
+                specificGroupStrategy.getTotalStCeloInSpecificGroupStrategies();
+            uint256 supposedStCeloInActiveGroup = stCeloInDefaultStrategy /
+                defaultStrategy.getGroupsLength();
+            uint256 supposedstCeloInSpecificGroupStrategy = specificGroupStrategy
+                .getTotalStCeloVotesForStrategy(group);
+
+            return (
+                manager.toCelo(supposedStCeloInActiveGroup + supposedstCeloInSpecificGroupStrategy),
+                realCelo
+            );
+        }
+
+        revert GroupNotEligible(group);
```

Comment:
When would this ever apply? We already returned with (0, real) for groups that aren't voted on by either strategy.

Reply:
Thanks, removed :-)